lint failing

<!--
BUGS: Please use this template.

SUPPORT REQUESTS/QUESTIONS: If you have a support request or question please
submit them to StackOverflow using the tags aurelia and aurelia-framework
http://stackoverflow.com/questions/tagged/aurelia
or the Aurelia Gitter https://gitter.im/aurelia/discuss

Blue Spire offers paid support agreements. Further information regarding paid support
may be obtained by emailing support@bluespire.com

Future support requests submitted here will be closed.

The HTML comments below are for your reference, and are not displayed
when your issue is submitted, feel free to leave them.

Choose one of the two headings, delete the other.
-->
**I'm submitting a bug report**

* **Library Version:**
0.1.16

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
7.7.4

* **NPM Version:**
4.1.2

* **Language:**
TypeScript 2.2.1


**Current behavior:**
lint operation errors with the following output

```
> aurelia-ux@0.1.16 lint E:\Development\Aurelia\ux
> cross-env tslint --project tsconfig.json

Warning: Got empty options

E:/Development/Aurelia/ux/src/hosts/cordova.ts[35, 9]: Identifier 'device' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/hosts/electron.ts[14, 9]: Identifier 'p' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-controller.ts[52, 27]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-controller.ts[23, 9]: Identifier 'overrideContext' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-controller.ts[24, 9]: Identifier '$styles' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-factory.ts[37, 38]: Parentheses are required around the parameters of an arrow function definition
E:/Development/Aurelia/ux/src/styles/style-factory.ts[61, 1]: A maximum of 1 class per file is allowed
E:/Development/Aurelia/ux/src/styles/style-factory.ts[31, 11]: Identifier 'baseStyles' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-factory.ts[27, 9]: Identifier '$styles' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-factory.ts[28, 9]: Identifier 'ux' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-factory.ts[13, 40]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/style-factory.ts[13, 96]: Don't use 'Object' as a type. Avoid using the `Object` type. Did you mean `object`?
E:/Development/Aurelia/ux/src/styles/style-compiler.ts[34, 1]: A maximum of 1 class per file is allowed
E:/Development/Aurelia/ux/src/styles/style-compiler.ts[42, 1]: A maximum of 1 class per file is allowed
E:/Development/Aurelia/ux/src/styles/style-compiler.ts[19, 22]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-compiler.ts[14, 11]: Identifier 'name' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-compiler.ts[12, 9]: Identifier 'styles' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-compiler.ts[13, 9]: Identifier 'transformed' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-compiler.ts[11, 35]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[112, 1]: A maximum of 1 class per file is allowed
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[147, 1]: A maximum of 1 class per file is allowed
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[19, 41]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[78, 30]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[93, 24]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[138, 24]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[163, 20]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[172, 23]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[42, 9]: Identifier 'quote' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[75, 11]: Identifier 'path' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[93, 13]: Identifier 'compiler' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[85, 9]: Identifier 'styleUrl' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[138, 13]: Identifier 'compiler' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[161, 9]: Identifier 'css' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[163, 9]: Identifier 'compiler' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[172, 9]: Identifier 'designType' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[34, 5]: Identifier 'cssUrlMatcher' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[13, 59]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[19, 29]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[73, 66]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[131, 66]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/style-strategy.ts[160, 66]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/style-locator.ts[28, 8]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-locator.ts[41, 10]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-locator.ts[50, 20]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-locator.ts[20, 11]: Identifier 'origin' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-locator.ts[49, 9]: Identifier 'origin' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-locator.ts[82, 9]: Identifier 'moduleId' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-locator.ts[83, 9]: Identifier 'id' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[22, 30]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[29, 25]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[30, 10]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[51, 25]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[59, 8]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[70, 23]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[76, 10]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[78, 10]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[21, 9]: Identifier 'themeKey' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[22, 9]: Identifier 'currentController' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[73, 11]: Identifier 'shadowDOMRoot' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[86, 9]: Identifier 'root' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-engine.ts[20, 57]: Don't use 'Object' as a type. Avoid using the `Object` type. Did you mean `object`?
E:/Development/Aurelia/ux/src/styles/style-resource.ts[34, 81]: Parentheses are required around the parameters of an arrow function definition
E:/Development/Aurelia/ux/src/styles/style-resource.ts[42, 1]: A maximum of 1 class per file is allowed
E:/Development/Aurelia/ux/src/styles/style-resource.ts[27, 26]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/styles/style-resource.ts[27, 9]: Identifier 'styleStrategy' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/styles/style-resource.ts[9, 27]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/style-resource.ts[16, 51]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/decorators.ts[19, 19]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/styles/dynamic-styles.ts[10, 16]: Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.
E:/Development/Aurelia/ux/src/ux-configuration.ts[34, 17]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/ux-configuration.ts[42, 17]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/ux-configuration.ts[34, 9]: Identifier 'proto' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/ux-configuration.ts[35, 9]: Identifier 'original' is never reassigned; use 'const' instead of 'let'.
src/aurelia-ux.ts[28, 42]: Parentheses are required around the parameters of an arrow function definition
src/aurelia-ux.ts[36, 41]: Parentheses are required around the parameters of an arrow function definition
src/aurelia-ux.ts[28, 9]: Identifier 'found' is never reassigned; use 'const' instead of 'let'.
src/index.ts[22, 12]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
src/index.ts[22, 7]: Identifier 'ux' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/designs/design-attributes.ts[16, 11]: Identifier 'realName' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/designs/design-attributes.ts[13, 9]: Identifier 'current' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/designs/design-attributes.ts[12, 19]: Identifier 'ii' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/designs/design-attributes.ts[9, 7]: Identifier 'prefix' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/designs/design-attributes.ts[10, 7]: Identifier 'length' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/element-rect.ts[78, 1]: Exceeds maximum line length of 120
E:/Development/Aurelia/ux/src/effects/element-rect.ts[81, 5]: Multiple variable declarations in the same statement are forbidden
E:/Development/Aurelia/ux/src/effects/element-rect.ts[85, 7]: variable name must be in camelcase or uppercase
E:/Development/Aurelia/ux/src/effects/element-rect.ts[86, 7]: variable name must be in camelcase or uppercase
E:/Development/Aurelia/ux/src/effects/element-rect.ts[81, 9]: Identifier 'l' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/element-rect.ts[82, 7]: Identifier 't' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/element-rect.ts[83, 7]: Identifier 'w' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/element-rect.ts[84, 7]: Identifier 'h' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/element-rect.ts[85, 7]: Identifier '_x' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/element-rect.ts[86, 7]: Identifier '_y' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[5, 1]: Multiple variable declarations in the same statement are forbidden
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[158, 5]: Multiple variable declarations in the same statement are forbidden
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[226, 5]: Multiple variable declarations in the same statement are forbidden
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[268, 5]: Multiple variable declarations in the same statement are forbidden
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[5, 5]: variable name must be in camelcase or uppercase
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[6, 3]: variable name must be in camelcase or uppercase
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[7, 3]: variable name must be in camelcase or uppercase
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[158, 9]: Identifier 'radius' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[162, 7]: Identifier 'elapsed' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[163, 7]: Identifier 'currentTime' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[164, 7]: Identifier 'actualRadius' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[226, 9]: Identifier 'translateFraction' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[268, 9]: Identifier 'scaleFactor' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[269, 7]: Identifier 'containerCenter' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[270, 7]: Identifier 'currentPos' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[271, 7]: Identifier 'deltaPos' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[301, 9]: Identifier 'containerCenter' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[5, 5]: Identifier '_window' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[6, 3]: Identifier '_doc' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-wave.ts[7, 3]: Identifier '_now' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[109, 28]: Parentheses are required around the parameters of an arrow function definition
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[165, 24]: Parentheses are required around the parameters of an arrow function definition
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[113, 9]: variable name must be in camelcase or uppercase
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[120, 9]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[127, 9]: Type assertion using the '<>' syntax is forbidden. Use the 'as' syntax instead.
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[113, 9]: Identifier '_doc' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[140, 9]: Identifier 'wave' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[154, 9]: Identifier 'wave' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[176, 9]: Identifier 'waveIndex' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[193, 11]: Identifier 'wave' is never reassigned; use 'const' instead of 'let'.
E:/Development/Aurelia/ux/src/effects/paper-ripple.ts[192, 21]: Identifier 'l' is never reassigned; use 'const' instead of 'let'.
src/input/ux-input.ts[53, 11]: Identifier 'attributeValue' is never reassigned; use 'const' instead of 'let'.
src/input/ux-input.ts[62, 11]: Identifier 'attributeValue' is never reassigned; use 'const' instead of 'let'.
src/input/ux-input.ts[102, 9]: Identifier 'blurEvent' is never reassigned; use 'const' instead of 'let'.
src/input/ux-input.ts[115, 9]: Identifier 'blurEvent' is never reassigned; use 'const' instead of 'let'.
src/textarea/ux-textarea.ts[52, 11]: Identifier 'attributeValue' is never reassigned; use 'const' instead of 'let'.
src/textarea/ux-textarea.ts[93, 9]: Identifier 'blurEvent' is never reassigned; use 'const' instead of 'let'.
src/textarea/ux-textarea.ts[106, 9]: Identifier 'blurEvent' is never reassigned; use 'const' instead of 'let'.
```

```
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "lint"
npm ERR! node v7.7.4
npm ERR! npm  v4.1.2
npm ERR! code ELIFECYCLE
npm ERR! aurelia-ux@0.1.16 lint: `cross-env tslint --project tsconfig.json`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the aurelia-ux@0.1.16 lint script 'cross-env tslint --project tsconfig.json'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the aurelia-ux package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     cross-env tslint --project tsconfig.json
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs aurelia-ux
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls aurelia-ux
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     E:\Development\Aurelia\ux\npm-debug.log
```

**Expected/desired behavior:**
<!--
If the current behavior is a bug, please provide the steps to reproduce and, if possible, a minimal demo of the
problem along with a runnable gist, if possible.
To create a runnable gist, go to https://gist.run/?id=7542e061bc940cde506b
Then click "Fork to Public Gist".
Create your gist, then finally click "Update Gist." Include a link to the gist.run below.
-->

For `npm run lint` to pass

--- a/src/aurelia-ux.ts
+++ b/src/aurelia-ux.ts
@@ -25,7 +25,7 @@ export class AureliaUX {
   }
 
   public start(config: FrameworkConfiguration) {
-    let found = this.availableHosts.find(x => x.isAvailable);
+    const found = this.availableHosts.find(x => x.isAvailable);
 
     if (found === undefined) {
       throw new Error('Could not determine host environment');

--- a/src/designs/design-attributes.ts
+++ b/src/designs/design-attributes.ts
@@ -6,17 +6,19 @@ export function processDesignAttributes(
   node: Element,
   attributes: NamedNodeMap,
   ___: any) {
-  let prefix = 'material-'; // TODO: get this from somewhere
-  let length = prefix.length;
+  const prefix = 'material-'; // TODO: get this from somewhere
+  const length = prefix.length;
 
+  // tslint:disable:prefer-const
   for (let i = 0, ii = attributes.length; i < ii; ++i) {
-    let current = attributes[i];
+    const current = attributes[i];
 
     if (current.name.indexOf(prefix) === 0) {
-      let realName = current.name.substring(length);
+      const realName = current.name.substring(length);
       node.setAttribute(realName, current.value);
     }
   }
+  // tslint:enable:prefer-const
 
   return true;
 }

--- a/src/effects/element-rect.ts
+++ b/src/effects/element-rect.ts
@@ -75,16 +75,17 @@ export class ElementRect {
    * @param {Number} ev.y - The `y` coordinate of the point.
    * @param {Number} ev.clientX - The `x` coordinate of the point.
    * @param {Number} ev.clientY - The `y` coordinate of the point.
-   * @returns {Boolean} Returns `true` if the `x` and `y` coordinates of point is a point inside this element's rectangle, otherwise `false`.
+   * @returns {Boolean} Returns `true` if the `x` and `y` coordinates of point is a
+   * point inside this element's rectangle, otherwise `false`.
    */
   public contains(ev: any) {
-    let l = this.boundingRect.left,
-      t = this.boundingRect.top,
-      w = this.boundingRect.width,
-      h = this.boundingRect.height,
-      _x = ev.x || ev.clientX || 0,
-      _y = ev.y || ev.clientY || 0;
+    const l = this.boundingRect.left;
+    const t = this.boundingRect.top;
+    const w = this.boundingRect.width;
+    const h = this.boundingRect.height;
+    const x = ev.x || ev.clientX || 0;
+    const y = ev.y || ev.clientY || 0;
 
-    return _x >= l && _x <= l + w && _y >= t && _y <= t + h;
+    return x >= l && x <= l + w && y >= t && y <= t + h;
   }
 }

--- a/src/effects/paper-ripple.ts
+++ b/src/effects/paper-ripple.ts
@@ -110,21 +110,21 @@ export class PaperRipple {
   }
 
   private initTarget(target: HTMLElement | null = null) {
-    let _doc: Document = PLATFORM.global.document;
+    const doc: Document = PLATFORM.global.document;
 
-    this.$ = target || _doc.createElement('div');
+    this.$ = target || doc.createElement('div');
     this.$.classList.add('paper-ripple');
 
     if (!this.$background) {
       this.$background = target &&
-        <HTMLElement>target.querySelector('.paper-ripple__background') || _doc.createElement('div');
+        target.querySelector('.paper-ripple__background') as HTMLElement || doc.createElement('div');
       this.$background.classList.add('paper-ripple__background');
       this.$.appendChild(this.$background);
     }
 
     if (!this.$waves) {
       this.$waves = target &&
-        <HTMLElement>target.querySelector('.paper-ripple__waves') || _doc.createElement('div');
+        target.querySelector('.paper-ripple__waves') as HTMLElement || doc.createElement('div');
       this.$waves.classList.add('paper-ripple__waves');
       this.$.appendChild(this.$waves);
     }
@@ -137,7 +137,7 @@ export class PaperRipple {
    * @returns Current instance for method chaining.
    */
   public addWave() {
-    let wave = new PaperWave(this);
+    const wave = new PaperWave(this);
     this.$waves.appendChild(wave.$);
     this.$background.style.backgroundColor = wave.color;
     this.waves.push(wave);
@@ -151,7 +151,7 @@ export class PaperRipple {
    * @returns Current instance for method chaining.
    */
   public downAction(ev: MouseEvent) {
-    let wave = this.addWave();
+    const wave = this.addWave();
     wave.downAction(ev);
     this.animate();
     return this;
@@ -173,7 +173,7 @@ export class PaperRipple {
    * @returns Current instance for method chaining.
    */
   public removeWave(wave: PaperWave) {
-    let waveIndex = this.waves.indexOf(wave);
+    const waveIndex = this.waves.indexOf(wave);
 
     if (waveIndex < 0) {
       return this;
@@ -189,8 +189,9 @@ export class PaperRipple {
    * @returns Current instance for method chaining.
    */
   public animate() {
+    // tslint:disable:prefer-const
     for (let i = 0, l = this.waves.length; i < l; i++) {
-      let wave = this.waves[i];
+      const wave = this.waves[i];
 
       if (wave) {
         wave.draw();
@@ -202,6 +203,7 @@ export class PaperRipple {
         }
       }
     }
+    // tslint:enable:prefer-const
 
     if (!this.shouldKeepAnimating && this.waves.length === 0) {
       this.$background.style.backgroundColor = null;

--- a/src/effects/paper-wave.ts
+++ b/src/effects/paper-wave.ts
@@ -2,9 +2,11 @@ import { PLATFORM } from 'aurelia-pal';
 import { ElementRect } from './element-rect';
 import { Point } from './point';
 
-let _window: Window = PLATFORM.global,
-  _doc = _window.document,
-  _now = PLATFORM.performance.now.bind(PLATFORM.performance);
+// tslint:disable:variable-name
+const _window: Window = PLATFORM.global;
+const  _doc = _window.document;
+const  _now = PLATFORM.performance.now.bind(PLATFORM.performance);
+// tslint:enable:variable-name
 
 /**
  * Provides all the logic to produce a one-time rippling effect.
@@ -155,13 +157,13 @@ export class PaperWave {
    * @returns {Number} The value of the wave's radius.
    */
   get radius() {
-    let radius = Math.min(
+    const radius = Math.min(
       Math.sqrt(Math.pow(this.containerRect.width, 2) + Math.pow(this.containerRect.height, 2)),
       PaperWave.MAX_RADIUS
-    ) * 1.1 + 5,
-      elapsed = 1.1 - 0.2 * (radius / PaperWave.MAX_RADIUS),
-      currentTime = this.mouseInteractionSeconds / elapsed,
-      actualRadius = radius * (1 - Math.pow(80, -currentTime));
+    ) * 1.1 + 5;
+    const elapsed = 1.1 - 0.2 * (radius / PaperWave.MAX_RADIUS);
+    const currentTime = this.mouseInteractionSeconds / elapsed;
+    const actualRadius = radius * (1 - Math.pow(80, -currentTime));
 
     return Math.abs(actualRadius);
   }
@@ -223,9 +225,9 @@ export class PaperWave {
    * @returns {{x: Number, y: Number}} Object containing coordinates of the wave's current position.
    */
   get currentPosition() {
-    let translateFraction = this.translationFraction,
-      x = this.startPosition.x,
-      y = this.startPosition.y;
+    const translateFraction = this.translationFraction;
+    let x = this.startPosition.x;
+    let y = this.startPosition.y;
 
     if (this.endPosition.x) {
       x = this.startPosition.x + translateFraction * (this.endPosition.x - this.startPosition.x);
@@ -265,10 +267,10 @@ export class PaperWave {
    * @returns Current instance for method chaining.
    */
   public draw() {
-    let scaleFactor = this.radius / (this.containerRect.size / 2),
-      containerCenter = this.containerRect.center,
-      currentPos = this.currentPosition,
-      deltaPos = {
+    const scaleFactor = this.radius / (this.containerRect.size / 2);
+    const containerCenter = this.containerRect.center;
+    const currentPos = this.currentPosition;
+    const deltaPos = {
         x: currentPos.x - containerCenter.x,
         y: currentPos.y - containerCenter.y
       };
@@ -298,7 +300,7 @@ export class PaperWave {
    * @returns Current instance for method chaining.
    */
   public downAction(event: MouseEvent | null = null) {
-    let containerCenter = this.containerRect.center;
+    const containerCenter = this.containerRect.center;
 
     this.resetDefaults();
 

--- a/src/hosts/cordova.ts
+++ b/src/hosts/cordova.ts
@@ -32,7 +32,7 @@ export class Cordova implements Host {
   }
 
   private getPlatformType() {
-    let device = PLATFORM.global.device || {platform: 'android'};
+    const device = PLATFORM.global.device || {platform: 'android'};
     return device.platform.toLowerCase();
   }
 }

--- a/src/hosts/electron.ts
+++ b/src/hosts/electron.ts
@@ -11,7 +11,7 @@ export class Electron implements Host {
   public type = 'electron';
 
   public get isAvailable() {
-    let p = PLATFORM.global.process;
+    const p = PLATFORM.global.process;
     return p && p.versions && p.versions.electron;
   }
 

--- a/src/index.ts
+++ b/src/index.ts
@@ -19,7 +19,7 @@ export function configure(config: FrameworkConfiguration, callback?: (config: Au
     './textarea/ux-textarea'
   ]);
 
-  let ux = <AureliaUX>config.container.get(AureliaUX);
+  const ux = config.container.get(AureliaUX) as AureliaUX;
 
   if (typeof callback === 'function') {
     return Promise.resolve(callback(ux))

--- a/src/input/ux-input.ts
+++ b/src/input/ux-input.ts
@@ -50,7 +50,7 @@ export class UxInput implements Themable {
     }
 
     if (this.element.hasAttribute('placeholder')) {
-      let attributeValue = this.element.getAttribute('placeholder');
+      const attributeValue = this.element.getAttribute('placeholder');
 
       if (attributeValue) {
         this.textbox.setAttribute('placeholder', attributeValue);
@@ -59,7 +59,7 @@ export class UxInput implements Themable {
     }
 
     if (this.element.hasAttribute('step')) {
-      let attributeValue = this.element.getAttribute('step');
+      const attributeValue = this.element.getAttribute('step');
 
       if (attributeValue) {
         this.textbox.setAttribute('step', attributeValue);
@@ -99,7 +99,7 @@ export class UxInput implements Themable {
   }
 
   public attached() {
-    let blurEvent = DOM.createCustomEvent('blur', { bubbles: true });
+    const blurEvent = DOM.createCustomEvent('blur', { bubbles: true });
 
     this.textbox.addEventListener('focus', () => {
       this.element.classList.add('focused');
@@ -112,7 +112,7 @@ export class UxInput implements Themable {
   }
 
   public detached() {
-    let blurEvent = DOM.createCustomEvent('blur', { bubbles: true });
+    const blurEvent = DOM.createCustomEvent('blur', { bubbles: true });
 
     this.textbox.removeEventListener('focus', () => {
       this.element.classList.add('focused');

--- a/src/styles/style-compiler.ts
+++ b/src/styles/style-compiler.ts
@@ -9,17 +9,17 @@ export class StyleCompiler {
   constructor(private bindingLanguage: BindingLanguage, private viewResources: ViewResources) {}
 
   public compile(styleObjectType: Function, css: string): StyleFactory {
-    let styles = Object.create(null);
-    let transformed = css.replace(classMatcher, (_: string, capture: string) => {
-      let name = capture.replace(/\-/g, '_');
+    const styles = Object.create(null);
+    const transformed = css.replace(classMatcher, (_: string, capture: string) => {
+      const name = capture.replace(/\-/g, '_');
       styles[name] = true;
       return '.${$styles.' + name + ' & oneTime}';
     });
 
-    let expression = <any>this.bindingLanguage.inspectTextContent(
+    let expression = this.bindingLanguage.inspectTextContent(
       this.viewResources,
       transformed
-    );
+    ) as any;
 
     if (expression === null) {
       expression = new PlainCSSBindingExpression(transformed);

--- a/src/styles/style-controller.ts
+++ b/src/styles/style-controller.ts
@@ -20,8 +20,8 @@ export class StyleController {
   }
 
   public bind(view: View) {
-    let overrideContext: any = view.overrideContext;
-    let $styles = overrideContext['$styles'] || {};
+    const overrideContext: any = view.overrideContext;
+    const $styles = overrideContext['$styles'] || {};
 
     overrideContext['$' + this.factory.themeKey] = this.bindingContext;
     overrideContext['$design'] = this.overrideContext.$design;
@@ -49,7 +49,7 @@ export class StyleController {
 
   private ensureStyleElementIsAddedToDocument() {
     if (this.styleElement === undefined) {
-      this.styleElement = <HTMLStyleElement>DOM.injectStyles('', this.destination);
+      this.styleElement = DOM.injectStyles('', this.destination) as HTMLStyleElement;
       this.bindingInstance = this.expression.createBinding(this.styleElement);
     } else if (!this.styleElement.parentNode) {
       this.styleElementParent.appendChild(this.styleElement);

--- a/src/styles/style-engine.ts
+++ b/src/styles/style-engine.ts
@@ -17,17 +17,17 @@ export class StyleEngine {
     return camelCase(Origin.get(obj.constructor).moduleMember + 'Theme');
   }
 
-  public applyTheme(themable: Themable, theme: string | Object | null) {
-    let themeKey = this.getThemeKeyForComponent(themable);
-    let currentController = (<any>themable.view)[themeKey];
+  public applyTheme(themable: Themable, theme: string | object | null) {
+    const themeKey = this.getThemeKeyForComponent(themable);
+    const currentController = (themable.view as any)[themeKey];
     let bindingContext: any;
     let newController: StyleController | undefined;
 
     if (!theme) {
       if (currentController !== currentController.factory.defaultController) {
         currentController.unbind();
-        newController = <StyleController>currentController.factory.defaultController;
-        (<any>themable.view)[themeKey] = newController;
+        newController = currentController.factory.defaultController as StyleController;
+        (themable.view as any)[themeKey] = newController;
         newController.bind(themable.view);
       }
 
@@ -48,15 +48,15 @@ export class StyleEngine {
       newController = this.controllers.get(bindingContext);
 
       if (!newController) {
-        newController = <StyleController>currentController.factory.create(
+        newController = currentController.factory.create(
           this.container,
           null,
           bindingContext
-        );
+        ) as StyleController;
       }
 
       currentController.unbind();
-      (<any>themable.view)[themeKey] = newController;
+      (themable.view as any)[themeKey] = newController;
       newController.bind(themable.view);
       this.controllers.set(bindingContext, newController);
 
@@ -67,15 +67,15 @@ export class StyleEngine {
   }
 
   public getOrCreateStyleController(view: View, factory: StyleFactory): StyleController {
-    let controller = (<any>view)[factory.themeKey];
+    let controller = (view as any)[factory.themeKey];
 
     if (controller === undefined) {
-      let shadowDOMRoot = this.getShadowDOMRoot(view);
+      const shadowDOMRoot = this.getShadowDOMRoot(view);
 
       if (shadowDOMRoot === null) {
-        (<any>view)[factory.themeKey] = controller = factory.getOrCreateDefault(this.container);
+        (view as any)[factory.themeKey] = controller = factory.getOrCreateDefault(this.container);
       } else {
-        (<any>view)[factory.themeKey] = controller = factory.create(view.container, shadowDOMRoot);
+        (view as any)[factory.themeKey] = controller = factory.create(view.container, shadowDOMRoot);
       }
     }
 
@@ -83,7 +83,7 @@ export class StyleEngine {
   }
 
   private getShadowDOMRoot(view: View) {
-    let root = view.container.get(DOM.boundary);
+    const root = view.container.get(DOM.boundary);
 
     if (root && root.host instanceof Element) {
       return root;

--- a/src/styles/style-factory.ts
+++ b/src/styles/style-factory.ts
@@ -10,7 +10,7 @@ export class StyleFactory {
   public themeKey: string;
   private defaultController: StyleController;
 
-  constructor(private styleObjectType: Function, private styles: string[], private expression: Object) {
+  constructor(private styleObjectType: Function, private styles: string[], private expression: object) {
     this.themeKey = camelCase(Origin.get(styleObjectType).moduleMember);
   }
 
@@ -24,11 +24,11 @@ export class StyleFactory {
   }
 
   public create(container: Container, destination?: Element, bindingContext?: any): StyleController {
-    let $styles: any = {};
-    let ux = container.get(AureliaUX);
+    const $styles: any = {};
+    const ux = container.get(AureliaUX);
 
     if (bindingContext) {
-      let baseStyles = this.getOrCreateDefault(container).bindingContext;
+      const baseStyles = this.getOrCreateDefault(container).bindingContext;
       Object.setPrototypeOf(bindingContext, baseStyles);
     } else {
       bindingContext = container.get(this.styleObjectType);

--- a/src/styles/style-locator.ts
+++ b/src/styles/style-locator.ts
@@ -17,7 +17,7 @@ export class StyleLocator {
    */
   public getStyleStrategy(value: any): StyleStrategy {
     if (typeof value === 'object' && 'getStyleStrategy' in value) {
-      let origin = Origin.get(value.constructor);
+      const origin = Origin.get(value.constructor);
 
       value = value.getStyleStrategy();
 
@@ -25,7 +25,7 @@ export class StyleLocator {
         value = new RelativeStyleStrategy(value);
       }
 
-      (<any>styleStrategy).assert(value);
+      (styleStrategy as any).assert(value);
 
       if (origin.moduleId) {
         value.makeRelativeTo(origin.moduleId);
@@ -38,7 +38,7 @@ export class StyleLocator {
       value = new RelativeStyleStrategy(value);
     }
 
-    if ((<any>styleStrategy).validate(value)) {
+    if ((styleStrategy as any).validate(value)) {
       return value;
     }
 
@@ -46,8 +46,8 @@ export class StyleLocator {
       value = value.constructor;
     }
 
-    let origin = Origin.get(value);
-    let strategy = <StyleStrategy>metadata.get(StyleLocator.styleStrategyMetadataKey, value);
+    const origin = Origin.get(value);
+    let strategy = metadata.get(StyleLocator.styleStrategyMetadataKey, value) as StyleStrategy;
 
     if (!strategy) {
       if (!origin.moduleId) {
@@ -79,8 +79,8 @@ export class StyleLocator {
    * @return The view url.
    */
   public convertOriginToStyleUrl(origin: Origin): string {
-    let moduleId = origin.moduleId;
-    let id = (moduleId.endsWith('.js') || moduleId.endsWith('.ts'))
+    const moduleId = origin.moduleId;
+    const id = (moduleId.endsWith('.js') || moduleId.endsWith('.ts'))
       ? moduleId.substring(0, moduleId.length - 3)
       : moduleId;
 

--- a/src/styles/style-resource.ts
+++ b/src/styles/style-resource.ts
@@ -24,7 +24,7 @@ export class StyleResource {
   }
 
   public load(container: Container): Promise<StyleResource> {
-    let styleStrategy = (<StyleLocator>container.get(StyleLocator))
+    const styleStrategy = (container.get(StyleLocator) as StyleLocator)
       .getStyleStrategy(this.styleObjectType);
 
     if (!styleStrategy.moduleId) {

--- a/src/styles/style-strategy.ts
+++ b/src/styles/style-strategy.ts
@@ -16,7 +16,7 @@ export interface StyleStrategy {
 /**
  * Decorator: Indicates that the decorated class/object is a style strategy.
  */
-export const styleStrategy: Function = (<any>protocol).create('aurelia:style-strategy', {
+export const styleStrategy: Function = (protocol as any).create('aurelia:style-strategy', {
   validate(target: any): string | boolean {
     if (!(typeof target.loadStyleFactory === 'function')) {
       return 'Style strategies must implement: loadStyleFactory(): Promise<StyleFactory>';
@@ -31,7 +31,7 @@ export const styleStrategy: Function = (<any>protocol).create('aurelia:style-str
   }
 });
 
-let cssUrlMatcher = /url\((?!['"]data)([^)]+)\)/gi;
+const cssUrlMatcher = /url\((?!['"]data)([^)]+)\)/gi;
 
 function fixupCSSUrls(address: string, css: string) {
   if (typeof css !== 'string') {
@@ -39,7 +39,7 @@ function fixupCSSUrls(address: string, css: string) {
   }
 
   return css.replace(cssUrlMatcher, (_, p1) => {
-    let quote = p1.charAt(0);
+    const quote = p1.charAt(0);
 
     if (quote === '\'' || quote === '"') {
       p1 = p1.substr(1, p1.length - 2);
@@ -72,17 +72,17 @@ export class RelativeStyleStrategy implements StyleStrategy {
    */
   public loadStyleFactory(container: Container, styleObjectType: Function): Promise<StyleFactory> {
     if (this.absolutePath === null && this.moduleId) {
-      let path = resolveForDesign(this.pathOrDesignMap, container);
+      const path = resolveForDesign(this.pathOrDesignMap, container);
 
       if (!path) {
-        this.absolutePath = (<StyleLocator>container.get(StyleLocator))
+        this.absolutePath = (container.get(StyleLocator) as StyleLocator)
           .convertOriginToStyleUrl(new Origin(this.moduleId, 'default'));
       } else {
         this.absolutePath = relativeToFile(path, this.moduleId);
       }
     }
 
-    let styleUrl = this.absolutePath || resolveForDesign(this.pathOrDesignMap, container);
+    const styleUrl = this.absolutePath || resolveForDesign(this.pathOrDesignMap, container);
 
     return container.get(Loader)
       .loadText(styleUrl)
@@ -90,7 +90,7 @@ export class RelativeStyleStrategy implements StyleStrategy {
       .then((text: string) => {
         text = fixupCSSUrls(styleUrl, text);
         this.css = text;
-        let compiler = <StyleCompiler>container.get(StyleCompiler);
+        const compiler = container.get(StyleCompiler) as StyleCompiler;
         return compiler.compile(styleObjectType, this.css);
       });
   }
@@ -135,7 +135,7 @@ export class ConventionalStyleStrategy implements StyleStrategy {
       .then((text: string) => {
         text = fixupCSSUrls(this.styleUrl, text);
         this.css = text;
-        let compiler = <StyleCompiler>container.get(StyleCompiler);
+        const compiler = container.get(StyleCompiler) as StyleCompiler;
         return compiler.compile(styleObjectType, this.css);
       });
   }
@@ -158,9 +158,9 @@ export class InlineStyleStrategy implements StyleStrategy {
    * Loads a style factory.
    */
   public loadStyleFactory(container: Container, styleObjectType: Function): Promise<StyleFactory> {
-    let css = resolveForDesign(this.cssOrDesignMap, container);
+    const css = resolveForDesign(this.cssOrDesignMap, container);
     this.transformedCSS = fixupCSSUrls(this.moduleId, css);
-    let compiler = <StyleCompiler>container.get(StyleCompiler);
+    const compiler = container.get(StyleCompiler) as StyleCompiler;
     return Promise.resolve(compiler.compile(styleObjectType, this.transformedCSS));
   }
 }
@@ -169,7 +169,7 @@ function resolveForDesign(valueOrDesignMap: string | any, container: Container):
   if (typeof valueOrDesignMap === 'string') {
     return valueOrDesignMap;
   } else {
-    let designType = (<AureliaUX>container.get(AureliaUX)).design.type;
+    const designType = (container.get(AureliaUX) as AureliaUX).design.type;
     return valueOrDesignMap[designType];
   }
 }

--- a/src/textarea/ux-textarea.ts
+++ b/src/textarea/ux-textarea.ts
@@ -49,7 +49,7 @@ export class UxTextarea implements Themable {
     }
 
     if (this.element.hasAttribute('placeholder')) {
-      let attributeValue = this.element.getAttribute('placeholder');
+      const attributeValue = this.element.getAttribute('placeholder');
 
       if (attributeValue) {
         this.textbox.setAttribute('placeholder', attributeValue);
@@ -90,7 +90,7 @@ export class UxTextarea implements Themable {
   }
 
   public attached() {
-    let blurEvent = DOM.createCustomEvent('blur', { bubbles: true });
+    const blurEvent = DOM.createCustomEvent('blur', { bubbles: true });
 
     this.textbox.addEventListener('focus', () => {
       this.element.classList.add('focused');
@@ -103,7 +103,7 @@ export class UxTextarea implements Themable {
   }
 
   public detached() {
-    let blurEvent = DOM.createCustomEvent('blur', { bubbles: true });
+    const blurEvent = DOM.createCustomEvent('blur', { bubbles: true });
 
     this.textbox.removeEventListener('focus', () => {
       this.element.classList.add('focused');

--- a/src/ux-configuration.ts
+++ b/src/ux-configuration.ts
@@ -31,15 +31,15 @@ export class UXConfiguration {
   }
 
   public commandHandler() {
-    let proto = <any>SyntaxInterpreter.prototype;
-    let original = proto.handleUnknownCommand;
+    const proto = SyntaxInterpreter.prototype as any;
+    const original = proto.handleUnknownCommand;
 
     /* tslint:disable:only-arrow-functions */
     proto.handleUnknownCommand = function(r: ViewResources, e: Element, i: any, ei?: any, c?: any) {
       if (i.attrName === 'styles') {
         i.attrName = 'class';
         i.attrValue = '$styles.' + i.command.replace(/\-/g, '_');
-        return (<any>this)['one-way'](r, e, i, ei, c);
+        return (this as any)['one-way'](r, e, i, ei, c);
       } else {
         return original.call(this, r, e, i, ei, c);
       }

--- a/tslint.json
+++ b/tslint.json
@@ -8,7 +8,12 @@
     "interface-name": [true, "never-prefix"],
     "no-shadowed-variable": false,
     "no-string-literal": false,
-    "trailing-comma": false,
-    "member-ordering": ["fields-first"]
+    "trailing-comma": [false],
+    "member-ordering": ["fields-first"],
+    "array-type": ["array-simple"],
+    "arrow-parens": false,
+    "max-classes-per-file": [false],
+    "prefer-for-of": false,
+    "ban-types": false
   }
 }

